include/boost/decimal/detail/cmath/cbrt.hpp

```diff
@@ -24,65 +24,149 @@ namespace decimal {
 namespace detail {
 
 template <typename T>
-constexpr auto cbrt_impl(T val) noexcept
+constexpr auto cbrt_impl(T x) noexcept
     BOOST_DECIMAL_REQUIRES(detail::is_decimal_floating_point_v, T)
 {
-    constexpr T zero {0, 0};
-    constexpr T one {1, 0};
+    const auto fpc = fpclassify(x);
 
     T result { };
 
-    if (isnan(val) || abs(val) == zero)
+    if ((fpc == FP_NAN) || (fpc == FP_ZERO))
     {
-        result = val;
-    }
-    else if (isinf(val))
-    {
-        if (signbit(val))
-        {
-            result = std::numeric_limits<T>::quiet_NaN();
-        }
-        else
-        {
-            result = val;
-        }
+        result = x;
     }
-    else if (val < zero)
+    else if (signbit(x))
     {
-        result = std::numeric_limits<T>::quiet_NaN();
+        result = -cbrt(-x);
     }
-    else if (val == one)
+    else if (fpc == FP_INFINITE)
     {
-        result = one;
+        result = std::numeric_limits<T>::infinity();
     }
     else
     {
-        constexpr T epsilon = std::numeric_limits<T>::epsilon() * 100;
-        T error = one / epsilon;
+        int exp10val { };
+
+        const auto gn { frexp10(x, &exp10val) };
 
-        T x {};
-        if (val > one)
+        const auto
+            zeros_removal
+            {
+                remove_trailing_zeros(gn)
+            };
+
+        const bool is_pure { static_cast<unsigned>(zeros_removal.trimmed_number) == 1U };
+
+        if(is_pure)
         {
-            // Scale down if val is large by dividing the exp by 3
-            int exp {};
-            auto sig = frexp10(val, &exp);
-            x = T{sig, exp / 3};
+            // Here, a pure power-of-10 argument gets a straightforward result.
+            // For argument 10^n where n is a multiple of 3, the result is exact.
+
+            const int p10 { exp10val + static_cast<int>(zeros_removal.number_of_removed_zeros) };
+
+            if (p10 == 0)
+            {
+                result = T { 1 };
+            }
+            else
+            {
+                const int p10_mod3 = (p10 % 3);
+                const int p10_div3 = (p10 / 3);
+
+                result = T { 1, p10_div3 };
+
+                switch (p10_mod3)
+                {
+                    case 2:
+                        result *= numbers::cbrt10_v<T>;
```

Comment:
`numbers::cbrt10_v` is new right? Can you please add it to the doc page on constants?